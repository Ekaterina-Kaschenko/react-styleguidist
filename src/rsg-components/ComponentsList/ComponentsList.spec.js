import React from 'react';
import { ComponentsListRenderer } from './ComponentsListRenderer';

const components = [
	{
		name: 'Button',
		slug: 'button',
	},
	{
		name: 'Input',
		slug: 'input',
	},
	{
		name: 'Textarea',
		slug: 'textarea',
	},
];

it('should render sections with nested components', () => {
	const actual = shallow(
		<ComponentsListRenderer items={components} classes={{}} />
	);

	expect(actual).toMatchSnapshot();
});

it('should return null when the list is empty', () => {
	const actual = shallow(
		<ComponentsListRenderer classes={{}} items={[]} />
	);

	expect(actual.node).toBe(null);
});
