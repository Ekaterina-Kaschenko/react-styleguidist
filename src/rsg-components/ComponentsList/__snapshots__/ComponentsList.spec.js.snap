exports[`test should render sections with nested components 1`] = `
<ul>
  <li
    className="">
    <_class
      href="#button">
      Button
    </_class>
  </li>
  <li
    className="">
    <_class
      href="#input">
      Input
    </_class>
  </li>
  <li
    className="">
    <_class
      href="#textarea">
      Textarea
    </_class>
  </li>
</ul>
`;
